### PR TITLE
Added support for Silvercrest coffee maker

### PR DESCRIPTION
**Pull request by popeen**
Added support for Silvercrest Coffee maker.

### CODE REVIEW

File: custom_components/tuya_local/devices/silvercrest_coffee_maker.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - id: 1
+      type: boolean
+      name: operation_mode
+      mapping:
+        - dps_val: false
+          value: "Off"
+        - dps_val: true
+          value: "On"
+    # Dummy dps, water_heater requires a temperature.
+    # The coffee maker has no sensor for this so we put in a fake dps for it.
```
Comment (make-all) [inline custom_components/tuya_local/devices/silvercrest_coffee_maker.yaml:18]: water_heater does not require temperature, do not add a dummy.

File: custom_components/tuya_local/devices/silvercrest_coffee_maker.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    name: Coffee Maker
+primary_entity:
+  entity: water_heater
+  icon: "mdi:coffee-maker"
+  dps:
+    - id: 1
+      type: boolean
+      name: operation_mode
+      mapping:
+        - dps_val: false
```
Comment (make-all) [inline custom_components/tuya_local/devices/silvercrest_coffee_maker.yaml:13]: It is better to use predefined operation modes from https://github.com/home-assistant/core/blob/dev/homeassistant/components/water_heater/strings.json#L11
"off" and "electric" are probably best (note all lower case)

File: custom_components/tuya_local/devices/silvercrest_coffee_maker.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      range:
+        min: 0
+        max: 0
+    - id: 4
+      type: string
+      name: current_state
+    - id: 12
+      type: string
+      name: keep_warm_min
+    - id: 101
```
Comment (make-all) [inline custom_components/tuya_local/devices/silvercrest_coffee_maker.yaml:33]: Does not need to be here if it is unused by the water_heater, and has a dedicated entity below.

File: custom_components/tuya_local/devices/silvercrest_coffee_maker.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+name: Silvercrest Coffee Maker
```
Comment (make-all) [inline custom_components/tuya_local/devices/silvercrest_coffee_maker.yaml:1]: This name should be swapped with the product specific name below.  Also, Home Assistant naming convention is to only capitalise the first word of names, unless trademarks or abbreviations require it.

**Comment by popeen** [comment]
Thanks for great feedback!

Made me go back and try switch again as specified by the data model and this time could make it work, I must have done something wrong the first time. Meaning it is no longer a water_heater. I am still not sure why I couldn't add the water meter without temperature but I'm sure it was something on my end.

I have changed the name and moved all the properties to separate entities. I think it should be ok now.

**Pull request merged**